src/components/LineChart/LineChart.tsx

```diff
@@ -0,0 +1,125 @@
+  labelX?: string;
+  /** Property indicating the label name to be displayed for Y axis */
+  labelY?: string;
+  /**  Define if the legend will be displayed */
+  showLegend?: boolean;
+  /** Function passed as property, so that the user can select colors for each line. If no function is passed or no color is picked for one or more lines, color will be selected randomly  */
+  color?: (dataLabel: string) => string;
+};
+
+const LineChart: React.FC<Props> = ({ data, labelX, labelY, showLegend = false, color }) => {
+  const theme = useTheme();
+
+  const uniqueKeyNames = useMemo(() => getKeyNames(data), [data]);
+
+  const colorsPicked = useMemo(
+    () => colorPicker({ theme, uniqueKeyNames, color }),
+
+    [theme, uniqueKeyNames, color]
+  );
+
+  const colors = Object.entries(colorsPicked);
+
+  return uniqueKeyNames.length <= Object.keys(theme.palette.flat).length ? (
+    <Wrapper data={data} margin={{ top: 10, right: 20, left: 20, bottom: 50 }}>
+      <defs>
+        {colors.map(([dataLabel, color]) => (
+          <GradientLine key={`color${dataLabel}`} dataLabel={dataLabel} color={color} />
+        ))}
+      </defs>
+      <XAxis
+        dataKey="name"
+        tickMargin={31}
+        axisLine={false}
+        tickLine={false}
+        label={labelX && { value: labelX, angle: 0, position: 'bottom', offset: 35 }}
+      />
+      <YAxis
+        tick={{ textAnchor: 'start' }}
+        tickMargin={40}
+        axisLine={false}
+        tickLine={false}
+        label={labelY && { value: labelY, angle: -90, position: 'left', offset: 15 }}
+      />
+      <CartesianGrid vertical={false} />
+      {showLegend && (
+        <Legend
+          align="left"
+          iconType="circle"
+          iconSize={16}
+          wrapperStyle={{
+            paddingTop: '50px',
+            paddingLeft: '13px',
+          }}
+        />
+      )}
+      <Tooltip
+        cursor={{ stroke: theme.utils.getColor('lightGray', 300), strokeWidth: 1 }}
+        content={<CustomTooltip />}
+      />
+      {colors.map(([dataLabel, color]) => (
+        <Area
+          key={dataLabel}
+          type="linear"
+          dataKey={dataLabel}
+          stroke={color}
+          fillOpacity={1}
+          fill={`url(#color${dataLabel})`}
+          activeDot={{ r: 3, stroke: 'none' }}
+          dot={{ r: 3, fill: color }}
+        />
+      ))}
+    </Wrapper>
+  ) : (
+    <p>You must define less than {Object.keys(theme.palette.flat).length} different data</p>
+  );
+};
+
+const Wrapper: React.FC<{
+  data: Data[];
+  margin: { top: number; right: number; left: number; bottom: number };
+}> = ({ data, children }) => {
+  if (process.env.NODE_ENV !== 'test') {
+    return (
+      <ResponsiveContainer aspect={1.5}>
+        <AreaChart data={data} margin={{ top: 10, right: 20, left: 20, bottom: 50 }}>
+          {children}
+        </AreaChart>
+      </ResponsiveContainer>
+    );
+  }
+
+  // for testing purposes only !!!
```

Comment:
We should make the chart's size responsive for real cases too.

Reply:
Not sure if i understand your comment correctly. We return `<ResponsiveContainer> 
<AreaChart/> 
</ResponsiveContainer>` only for real cases. On test cases we just return `<AreaChart/>` because rechart's ResponsiveContainer component was not working well with our tests.

Reply:
Testing a library is necessary as soon as the library does not cover a case by its tests. 

Besides that, you created a dependency on a system variable (NODE_ENV). We may face side effects by using this component on a third-party project.

Maybe we should consider removing the dummy component.